pkg/download/mode/mode.go

```diff
@@ -64,11 +69,16 @@ func NewFile(m Mode, downloadURL string) (*DownloadFile, error) {
 		}
 		return parseFile(bts)
 	}
+
+	if m == "" {
+		return nil, errors.E(op, ErrNoDownloadMode)
+	}
+
 	switch m {
 	case Sync, Async, Redirect, AsyncRedirect, None:
 		return &DownloadFile{Mode: m, DownloadURL: downloadURL}, nil
 	default:
-		return nil, errors.E(op, "unrecognized download mode: "+m, errors.KindBadRequest)
+		return nil, errors.E(op, fmt.Sprintf(invalidModeErr, m))
```

Comment:
let's still pass a `errors.KindBadRequest` just in case the caller in the future needs to check if there was a parsing error, or a user error

Reply:
Okay, I think I'll pass it as an argument _before_ the human-friendly string, so that the log message makes sense in this case

Reply:
@twexler the order doesn't matter, I think the reason the friendly message was not showing up is because of the fact that `"some string " + m` was actually a Mode type, and not a string that our errors package didn't understand. That's a very tiny bug for later :)

Reply:
Order does matter, if you look at this:
https://github.com/gomods/athens/blob/master/pkg/errors/errors.go#L91-L106 the _last_ error will get set to `e.Err`, which is then used in `Error.Error()` as the error string to return.

Reply:
The `errors.KindBadRequest` is an int, which will never get set as the *error*, but it will get set as the *Kind*. The problem with the confusing error message was happening here: https://github.com/gomods/athens/blob/master/pkg/errors/errors.go#L107:L109

So whether it comes first, or last shouldn't matter. 

That is because we never passed a `string` in the args, which a mistake in the codebase itself. I think a more accurate way to do it, is to log-out (or even panic) unrecognized types such as mode.Mode

Reply:
You're absolutely right, I thought those actually implemented `error`, my bad.

I'm happy to switch these errors to full on panics, although they become harder to test for.

Reply:
@twexler for now, let's leave the errors package and fix it in another PR. I think a `panic` should work, because it will call out bugs in our own codebase, even if it's reported by a user (unfortunately)

Reply:
Sounds good, I just pulled in the latest changes.